iocore/net/UnixNetVConnection.cc

```diff
@@ -1303,15 +1303,15 @@ UnixNetVConnection::connectUp(EThread *t, int fd)
   }
 
   if (check_emergency_throttle(con)) {
-    // The `con' could be closed if there is hyper emergency
-    if (con.fd == NO_FD) {
+    if (fd == NO_FD) {
+      con.close();
       // We need to decrement the stat because close_UnixNetVConnection only decrements with a valid connection descriptor.
       NET_SUM_GLOBAL_DYN_STAT(net_connections_currently_open_stat, -1);
       // Set errno force to EMFILE (reached limit for open file descriptors)
       errno = EMFILE;
       res   = -errno;
-      goto fail;
     }
+    goto fail;
```

Comment:
This should be clean up since the con.fd is not closed. The `net_connections_currently_open_stat` can be decreased by `free(t)`.

Reply:
Based on discussion, I need to open a new pr for this `net_connections_currently_open_stat`.